src/meta/name.rs

```diff
@@ -22,14 +21,18 @@ pub struct Name {
 
 impl Name {
     pub fn new(path: &Path, file_type: FileType) -> Self {
-        let name = match path.file_name() {
-            Some(name) => name.to_string_lossy().to_string(),
-            None => path.to_string_lossy().to_string(),
+        let full = path.to_string_lossy().to_string();
+
+        let name = if let Some(idx) = full.rfind('/') {
+            full.split_at(idx + 1).1.into()
+        } else {
+            full
         };
 
-        let extension = path
-            .extension()
-            .map(|ext| ext.to_string_lossy().to_string());
+        let extension = match name.rfind('.') {
```

Comment:
This too, probably not a good idea. Better let the language do this.
We will not be aware of all the edge cases related to extensions and it is really easy to mess things up.